File: Programming_5/Department.py
from Validation import Validation

#Клас ВІДДІЛ:
# ID, title, director_name, phone_number, monthly_budget, yearly_budget, website_url.
el = Validation()


class Department(object):
    def __init__(self, **array):
        for(key, value) in array.items():
            setattr(self, key, array.get(key, value))

    def __str__(self):
        result = ""
        for i in dir(self):
            if not i.startswith("__") and not i.startswith("_") and i != "input":
                result += i + ": " + str(getattr(self, i)) + "\n"
        return result

    @property
    def ID(self):
        return self._ID

    @ID.setter
    @Validation.decorator_number
    def ID(self, val):
        self._ID = val

    @property
    def title(self):
        return self._title

    @title.setter
    @Validation.decorator_string
    def title(self, val):
        self._title = val

    @property
    def director_name(self):
        return self._director

    @director_name.setter
    @Validation.decorator_string
    def director_name(self, val):
        self._director = val

    @property
    def phone_number(self):
        return self._phone_number

    @phone_number.setter
    @Validation.decorator_nomer
    def phone_number(self, val):
        self._phone_number = val

    @property
    def monthly_budget(self):
        return self._monthly_budget

    @monthly_budget.setter
    @Validation.decorator_price
    def monthly_budget(self, val):
        self._monthly_budget = val

    @property
    def yearly_budget(self):
        return self._yearly_budget

    @yearly_budget.setter
    @Validation.decorator_price
    def yearly_budget(self, val):
        self._yearly_budget = val

    @property
    def website(self):
        return self._website_url

    @website.setter
    @Validation.decorator_url
    def website(self, val):
        self._website_url = val


    def input(self):
       K = dict((i, input(i)) for i in dir(self)
                     if not i.startswith("__") and not i.startswith("_") and i != "input")
       for (key, value) in K.items():
           setattr(self, key, K.get(key, value))
       return K

File: C#/Classes/Classes/Department.cs
//Клас ВІДДІЛ: ID, title, director_name, phone_number, monthly_budget, yearly_budget, website_url.
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Classes
{

    class Department
    {
        private string _ID;
        private string _title;
        private string _director_name;
        private string _phone_number;
        private double _monthly_budget;
        private double _yearly_budget;
        private string _website;

        public string ID
        {
            get => _ID;
            set { _ID = Validation.Number(value); }
        }

        public string title
        {
            get => _title;
            set { _title = Validation.String(value); }
        }

        public string director_name
        {
            get => _director_name;
            set { _director_name = Validation.String(value); }
        }

        public string phone_number
        {
            get => _phone_number;
            set { _phone_number = Validation.Phone(value); }
        }

        public double monthly_budget
        {
            get => _monthly_budget;
            set { _monthly_budget = Validation.Price(value); }
        }

        public double yearly_budget
        {
            get => _yearly_budget;
            set
            {
                double el = Validation.Price(Convert.ToDouble(value));
                _yearly_budget = Validation.Price(monthly_budget, el);
            }
        }

        public string website
        {
            get => _website;
            set { _website = Validation.URL(value); }
        }

        public string print()
        {
            return ($"ID: {ID} \nTitle: {title} \nDirector_name: {director_name} \nPhone_number: {phone_number} \nMonthly_budget: {monthly_budget} " +
                $"\nYearly_budget: {yearly_budget} \nWebsite: {website}\n");
        }

        public void input()
        {
            Console.Write("ID: ");
            ID = Console.ReadLine();
            Console.Write("Title: ");
            title = Console.ReadLine();
            Console.Write("Director_name: ");
            director_name = Console.ReadLine();
            Console.Write("Phone_number: ");
            phone_number = Console.ReadLine();
            Console.Write("Monthly_budget: ");
            monthly_budget = int.Parse(Console.ReadLine());
            Console.Write("Yearly_budget: ");
            yearly_budget = int.Parse(Console.ReadLine());
            Console.Write("Website: ");
            website = Console.ReadLine();
        }
    }
}